ruby/ql/lib/codeql/ruby/dataflow/internal/DataFlowPrivate.qll

```diff
@@ -181,6 +181,12 @@ module LocalFlow {
         ) and
         nodeFrom.asExpr() = for.getValue()
       )
+    or
+    nodeTo.asExpr() =
```

Comment:
This is already included in taint tracking, and I don't think it is safe to include in data flow, as the value may not be preserved.

Reply:
Why wouldn't it be value preserving?

Reply:
As far as I know:
* `a || b`  means `if (tmp = a) then tmp else b`
* `a && b`  means `if (tmp = a) then b else tmp`

Reply:
You are right!

Reply:
I guess I should remove the equivalent code from the taint steps.